avm/src/vm/opcode.rs

```diff
@@ -3031,779 +3054,784 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       // MUST read these first in case the arguments are themselves closure args being overwritten
       // for the recursive function.
       // Since we mutate the `Self` object in this, it *must* be read as mutable *first* to make
       // sure that the later registration of the `Self` object is pointing at the correct copy
-      let slf = hm.read_mut_fractal(args[0]);
+      let slf = hm.read_mut_fractal(args[0])?;
       let recurse_fn = HandlerFragment::new(slf[1].1, 0);
       let seq_addr = slf[0].0;
       drop(slf);
-      hm.register(CLOSURE_ARG_MEM_START + 1, args[0], false);
-      hm.register(CLOSURE_ARG_MEM_START + 2, args[1], false);
-      let seq = hm.read_mut_fractal_by_idx(seq_addr);
+      hm.register(CLOSURE_ARG_MEM_START + 1, args[0], false)?;
+      hm.register(CLOSURE_ARG_MEM_START + 2, args[1], false)?;
+      let seq = hm.read_mut_fractal_by_idx(seq_addr)?;
       let curr = seq[0].1;
       if curr < seq[1].1 {
         seq[0].1 = curr + 1;
-        hm = recurse_fn.run(hm).await;
-        hm = hm.drop_parent();
+        hm = recurse_fn.run(hm).await?;
+        hm = hm.drop_parent()?;
         // CANNOT `join` the memory like usual because the nested `recurse` calls have set "future"
         // values in the handler and will cause weird behavior. Only transfer the Self mutation and
         // the return value between iterations
-        HandlerMemory::transfer(&mut hm, CLOSURE_ARG_MEM_START, &mut hand_mem, args[2]);
-        HandlerMemory::transfer(&mut hm, CLOSURE_ARG_MEM_START + 1, &mut hand_mem, args[0]);
+        HandlerMemory::transfer(&mut hm, CLOSURE_ARG_MEM_START, &mut hand_mem, args[2])?;
+        HandlerMemory::transfer(&mut hm, CLOSURE_ARG_MEM_START + 1, &mut hand_mem, args[0])?;
       } else {
-        hand_mem.init_fractal(args[2]);
-        hand_mem.push_fixed(args[2], 0);
-        hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal("error: sequence out-of-bounds"));
+        hand_mem.init_fractal(args[2])?;
+        hand_mem.push_fixed(args[2], 0)?;
+        hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal("error: sequence out-of-bounds"))?;
       }
-      hand_mem
+      Ok(hand_mem)
     })
   });
   cpu!(seqrec => fn(args, hand_mem) {
     if args[1] == NOP_ID {
-      panic!("NOP can't be recursive");
+      return Err(VMError::InvalidState(InvalidState::InvalidNOP));
     }
-    hand_mem.init_fractal(args[2]);
-    hand_mem.push_register(args[2], args[0]);
-    hand_mem.push_fixed(args[2], args[1]);
-    None
+    hand_mem.init_fractal(args[2])?;
+    hand_mem.push_register(args[2], args[0])?;
+    hand_mem.push_fixed(args[2], args[1])?;
+    Ok(None)
   });
 
   // "Special" opcodes
   cpu!(exitop => fn(args, hand_mem) {
-    io::stdout().flush().unwrap();
-    io::stderr().flush().unwrap();
-    std::process::exit(hand_mem.read_fixed(args[0]) as i32);
+    let _ = io::stdout().flush();
+    let _ = io::stderr().flush();
```

Comment:
I feel like these lines are missing `?`s

Reply:
I was thinking that we don't care about the `Result`s returned from flushing, but it does make sense to do so.